Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py

```diff
@@ -0,0 +1,607 @@
+        'Tactics': properties.get('tactics'),
+        'FirstActivityTimeGenerated': format_date(properties.get('firstActivityTimeGenerated')),
+        'LastActivityTimeGenerated': format_date(properties.get('lastActivityTimeGenerated')),
+        'Etag': inc_data.get('etag')
+    }
+    return formatted_data
+
+
+def get_update_incident_request_data(client, args):
+    # Get Etag and other mandatory properties (title, severity, status) for update_incident command
+    _, _, result = get_incident_by_id_command(client, args)
+
+    title = args.get('title')
+    description = args.get('description')
+    severity = args.get('severity')
+    status = args.get('status')
+
+    if not title:
+        title = result.get('properties', {}).get('title')
+    if not description:
+        description = result.get('properties', {}).get('description')
+    if not severity:
+        severity = result.get('properties', {}).get('severity')
+    if not status:
+        status = result.get('properties', {}).get('status')
+
+    inc_data = {
+        'etag': result.get('etag'),
+        'properties': {
+            'title': title,
+            'description': description,
+            'severity': severity,
+            'status': status
+        }
+    }
+    remove_nulls_from_dictionary(inc_data['properties'])
+
+    return inc_data
+
+
+def comment_data_to_demisto_format(comment_data, inc_id):
+    properties = comment_data.get('properties', {})
+
+    formatted_data = {
+        'ID': comment_data.get('name'),
+        'IncidentID': inc_id,
+        'Message': properties.get('message'),
+        'AuthorName': properties.get('author', {}).get('assignedTo'),
+        'AuthorEmail': properties.get('author', {}).get('email'),
+        'CreatedTimeUTC': format_date(properties.get('createdTimeUtc'))
+    }
+    return formatted_data
+
+
+def related_resource_data_to_demisto_format(resource_data, inc_id):
+    properties = resource_data.get('properties', {})
+
+    formatted_data = {
+        'ID': properties.get('relatedResourceName'),
+        'Kind': properties.get('relatedResourceKind'),
+        'IncidentID': inc_id
+    }
+    return formatted_data
+
+
+def flatten_entity_attributes(attributes):
+    # This method flattens a GET entity response json.
+    flattened_results = attributes.get('properties', {})
+    flattened_results['ID'] = attributes.get('name')
+    flattened_results['Kind'] = attributes.get('kind')
+    return flattened_results
+
+
+def severity_to_level(severity):
+    if severity == 'Informational':
+        return 0.5
+    elif severity == 'Low':
+        return 1
+    elif severity == 'Medium':
+        return 2
+    elif severity == 'High':
+        return 3
+    return 0
+
+
+''' INTEGRATION COMMANDS '''
+
+
+def test_module(client):
+    # todo: in the current method of authorization we can't test the instance - can't use the auth_code twice
+    list_incidents_command(client, {'top': '1'})
```

Comment:
update this to return a good error msg when it fails - will be possible once the Oproxy is done